apps/rich-list-api/src/modules/BlockchainCppModule.ts

```diff
@@ -0,0 +1,41 @@
+import { Global, Module } from '@nestjs/common'
+import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
+import { ConfigService } from '@nestjs/config'
+import { ApiClient } from '@defichain/jellyfish-api-core'
+import { ActuatorProbes, BlockchainCppProbeIndicator } from '@defichain-apps/libs/actuator'
+
+/**
+ * Ain Module configures and export JsonRpcClient connected to a DeFiD.
+ * This does not have any side effect it merely configures and export a JsonRpcClient.
+ */
+@Global()
+@Module({
+  providers: [
+    BlockchainCppProbeIndicator,
+    {
+      provide: ApiClient,
+      useFactory: (configService: ConfigService): ApiClient => {
+        const url = configService.get<string>('BLOCKCHAIN_CPP_URL')
+        if (url === undefined) {
+          throw new Error('BlockchainCppModule config:BLOCKCHAIN_CPP_URL not provided')
+        }
+        return new JsonRpcClient(configService.get<string>('BLOCKCHAIN_CPP_URL') as string)
+      },
+      inject: [ConfigService]
+    }
+  ],
+  exports: [
+    ApiClient
+  ]
+})
+export class BlockchainCppModule {
```

Comment:
is it possible to adjust it so its agnostic to these differences?

Reply:
hmm.. i think i got idea.. the easier one is to standardize their config properties
comparison below
ocean-api, rich-list-api
use:
BLOCKCHAIN_CPP_URL

pg-api
uses:
defid.url
defid.liveness.maxBlockCount // test
defid.readiness.maxBlockCount // test
playground.maxBlockCount

Reply:
yeah I think that is definitely the way to go, configuration should be based on the central source of truth which is the module itself. The implements should reference the defined config rather than create their own. 👍🏻

Reply:
There is the `Joi` config validation, naming should be taken care anyway, we can consider custom make a `ConfigModule` instead of directly using `ConfigService`.